### PR TITLE
[FIX] Issue #7 : Check products before fetching price

### LINKED ISSUE #7: Bad Request Error - USDC
Got a Bad Request Error since adding USDC-Coin

Log says:

16:57:08  1> Refreshing account Coinbase Pro.
          1> Web Banking Engine: Using user-supplied extension Coinbase Pro.lua version 1.00.
          1> Protocol version: Web Scraping
          1> Server address: https://api.pro.coinbase.com
          1> Getting portfolio for Coinbase Pro...
          1> Sending: GET https://api.pro.coinbase.com/accounts
          1> HTTPS response: Bad Request
          1> The server responded with the error message: Bad Request. Retrying...
          1> Sending: GET https://api.pro.coinbase.com/accounts
          1> HTTPS response: Bad Request
          1> The API server of your bank responded with the error message: »Bad Request«  Please try again later or contact the support of your bank with this error message.

### PR DESCRIPTION
**Pull request by TheSerapher**
As requested, a quick fix for the current issue with unavailable currencies in the ticker.

Fixes #7

### CODE REVIEW

File: Coinbase Pro.lua [hunk 1]
```diff
@@ -123,10 +130,16 @@ function queryCoinbaseProApi(endpoint)
         return JSON(content):dictionary()
 end
 
-function queryExchangeRates(currency)
-        local url = string.format("https://api.pro.coinbase.com/products/%s-%s/ticker", currency, nativeCurrency)
-        local content = Connection():request("GET", url)
-        return JSON(content):dictionary()
+function queryExchangeRate(product_id)
+        local content = Connection():request("GET", "https://api.pro.coinbase.com/products/" .. product_id .. "/ticker")
+        return JSON(content):dictionary()["price"]
```
Comment (qoomon) [inline Coinbase Pro.lua]:
```suggestion
        return queryCoinbaseProApi("products/" .. product_id .. "/ticker")["price"]
```

Comment (qoomon) [reply Coinbase Pro.lua]: Except of this I really like this solution, thx a lot

**Comment by qoomon** [comment]
LGTM

**Comment by qoomon** [comment]
I just got an answer from the moneymoney support how to do it the right way
you have to set the `Accept` header to `application/json` then no exception is thrown, even if the api response code is 404. e.g.
```lua
local headers = {}
headers["Accept"] = "application/json"
local content = Connection():request("GET", url, nil, nil, headers)
return JSON(content):dictionary()
```

File: Coinbase Pro.lua [hunk 2]
```diff
@@ -123,10 +130,15 @@ function queryCoinbaseProApi(endpoint)
         return JSON(content):dictionary()
 end
 
-function queryExchangeRates(currency)
-        local url = string.format("https://api.pro.coinbase.com/products/%s-%s/ticker", currency, nativeCurrency)
-        local content = Connection():request("GET", url)
-        return JSON(content):dictionary()
+function queryExchangeRate(product_id)
+        return queryCoinbaseProApi("products/" .. product_id .. "/ticker")["price"]
```
Comment (leoschweizer) [inline Coinbase Pro.lua:134]: This actually intentionally did not use queryCoinbaseProApi because the ticker requests don't need to be signed.

**Comment by leoschweizer** [comment]
@TheSerapher having that information from @qoomon now, would you like to implement this differently, or leave it as it is?

**Comment by TheSerapher** [comment]
I think instead of causing wild 404s and checking for headers and status codes, why not keep it as is and check if we should call it at all.

File: Coinbase Pro.lua (hunk 2 above)
Comment (TheSerapher) [reply Coinbase Pro.lua:134]: Even though they are public, I think using this should not be detrimental at all. At least it looks cleaner that way :D

**Pull request merged**